src/pkg/utils/wait.go

```diff
@@ -90,11 +90,16 @@ func ExecuteWait(ctx context.Context, waitTimeout, waitNamespace, condition, kin
 		select {
 		case <-expired:
 			return errors.New("wait timed out")
+		case <-ctx.Done():
+			return errors.New("received interrupt")
```

Comment:
Not blocking - curious if you believe propagation with `ctx.Err()` included in the error returned (or even checking for specific errors such as `context.Canceled would net any long-term value?

We don't use `context.WithTimeout` in this function so we wouldn't get the benefit of coupling and checking for them together.